src/jump_input.jl

```diff
@@ -0,0 +1,298 @@
+    )
+    return _build_single_model(upper, lower, linkLU, linkLOnly)
+end 
+
+
+function _build_single_model(
+    upper::MOI.ModelLike, 
+    lower::MOI.ModelLike, 
+
+    # A dictionary that maps variables in the upper to variables in the lower
+    # upper_to_lower_link::Dict{MOI.VariableIndex,MOI.VariableIndex},
+    lower_to_upper_link::Dict{MOI.VariableIndex, MOI.VariableIndex},
+    lower_only::Dict{MOI.VariableIndex, MOI.VariableIndex}
+)
+
+    # A new model to build
+    #model = MOI.Utilities.Model{Float64}()
+    model = MOI.FileFormats.MPS.Model()
+    
+    # Create a copy of the upper model
+    upper_to_model_link = MOI.copy_to(model, upper)
+
+    #upper_variables = [upper_to_model_link[k] for k in keys(upper_to_lower_link)]
+    lower_variables = [upper_to_model_link[k] for k in values(lower_only)]
+
+    
+    lower_constraints = Vector{MOI.ConstraintIndex}()
+    for (F, S) in MOI.get(lower, MOI.ListOfConstraints())
+        for ci in MOI.get(lower, MOI.ListOfConstraintIndices{F,S}())
+            
+                lower_f = MOI.get(lower, MOI.ConstraintFunction(), ci)
+                lower_s = MOI.get(lower, MOI.ConstraintSet(), ci)
+
+                lower_f = MOI.Utilities.map_indices(lower_f) do x
+                return upper_to_model_link[lower_to_upper_link[x]]
+                end
+                new_ci = MOI.add_constraint(model, lower_f, lower_s)
+
+            if F == MOI.ScalarAffineFunction{Float64}
+                push!(lower_constraints, new_ci)    
+            end
+        end
+    end
+ 
+    
+    lower_primal_obj = MOI.get(lower, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}()) 
+    lower_objective = MOI.Utilities.map_indices(lower_primal_obj) do x
+        return upper_to_model_link[lower_to_upper_link[x]]
+    end
+
+    lower_sense = MOI.get(lower, MOI.ObjectiveSense())
+
+    return model, lower_variables, lower_objective, lower_constraints, lower_sense
+end
+
+
+function index_to_row_link(
+    model::MOI.FileFormats.MPS.Model)
+
+    i = 0
+    dict = Dict{MOI.ConstraintIndex, Int}()
+    for (S, _) in MOI.FileFormats.MPS.SET_TYPES
+        for ci in MOI.get(
+            model,
+            MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{Float64}, S}(),
+        )  
+            dict[ci] = i
+            i += 1
+        end
+    end
+    return dict
+end
+
+
+function index_to_column_link(
+    model::MOI.FileFormats.MPS.Model)
+
+    variables = MOI.get(model, MOI.ListOfVariableIndices())
+    return Dict{MOI.VariableIndex,Int}(
+        x => i - 1 for (i, x) in MOI.enumerate(variables)
+    )
+end
+
+
+function write_auxillary_file(
+    new_model::MOI.FileFormats.MPS.Model,
+    lower_variables::Vector{MOI.VariableIndex},
+    lower_objective::MOI.ScalarAffineFunction,
+    lower_constraints::Vector{MOI.ConstraintIndex},
+    lower_sense::MOI.OptimizationSense,
+    aux_address::AbstractString
+    )
+
+    Row = index_to_row_link(new_model)
```

Comment:
```suggestion
    row = index_to_row_link(new_model)
```
Variables should be lower-case